### PR TITLE
Fixes #8206 -- Implement data insight refinement for entity report data

### LINKED ISSUE #8206: Create Data Processing Ingestion Workflow for data insight
## Context
Metrics and dimensions for data insight exist in a somewhat raw form in the platform (web analytic + entity data). In order to serve modeled data for consumption by the frontend we would like pre-aggregate data in a refined form

## Solution
- [ ] Pre aggregate data based on report definition from `v1/analytics/report` -- report definition will need to be created
- [ ] Implement workflow logic to execute the pre aggregation daily
- [ ] Implement indexation (ElasticSearch) and write logic (MySQL)

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
This PR implements:
- the workflow to refine data for the data insight
- the refinement logic for the `EntityReportData`
- The ingestion to elasticsearch of refined data

Sample documents
```
{
    "took": 3,
    "timed_out": false,
    "_shards": {
        "total": 1,
        "successful": 1,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 8,
            "relation": "eq"
        },
        "max_score": 1.0,
        "hits": [
            {
                "_index": "entity_report_data_index",
                "_type": "_doc",
                "_id": "uRNUD4QBt2ICbn0kCg7D",
                "_score": 1.0,
                "_source": {
                    "id": null,
                    "timestamp": 1666704476301,
                    "reportDataType": "EntityReportData",
                    "data": {
                        "entityType": "Chart",
                        "entityTier": null,
                        "team": null,
                        "organization": null,
                        "completedDescriptions": 12,
                        "missingDescriptions": 0,
                        "hasOwner": 0,
                        "missingOwner": 12,
                        "entityCount": 12
                    }
                }
            },
            {
                "_index": "entity_report_data_index",
                "_type": "_doc",
                "_id": "uhNUD4QBt2ICbn0kCg7V",
                "_score": 1.0,
                "_source": {
                    "id": null,
                    "timestamp": 1666704476301,
                    "reportDataType": "EntityReportData",
                    "data": {
                        "entityType": "Dashboard",
                        "entityTier": null,
                        "team": null,
                        "organization": null,
                        "completedDescriptions": 12,
                        "missingDescriptions": 0,
                        "hasOwner": 0,
                        "missingOwner": 12,
                        "entityCount": 12
                    }
                }
            },
            [...]
```

This will then be aggregated to return formatted values for the frontend to consume.

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
Ingestion: @open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/utils/workflow_output_handler.py [hunk 1]
```diff
@@ -242,3 +242,33 @@ def print_test_suite_status(workflow) -> None:
...
+
+
+def print_data_insight_status(workflow) -> None:
+    """Runs click echo to print the test suite workflow results
+
+    Args:
+        workflow (DataInsightWorkflow): workflow object
+    """
+    click.echo()
+    click.secho("Processor Status:", bold=True)
```
Comment (pmbrull) [inline ingestion/src/metadata/utils/workflow_output_handler.py:254]: we're moving away from click in https://github.com/open-metadata/OpenMetadata/pull/8182, let's not add it to new changes 🙏

File: openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
       "description": "Type of Pipeline - metadata, usage",
       "type": "string",
       "javaType": "org.openmetadata.schema.entity.services.ingestionPipelines.PipelineType",
-      "enum": ["metadata", "usage", "lineage", "profiler", "TestSuite"]
+      "enum": ["metadata", "usage", "lineage", "profiler", "TestSuite", "DataInsight"]
```
Comment (pmbrull) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json:14]: we need to come up with a single naming convention here, as we moved from `profiler` to `TestSuite` etc

**Review by pmbrull — APPROVED** [review]
Thanks @TeddyCr, putting the stamp on. Just left a few comments if you have the time

File: ingestion/src/metadata/data_insight/processor/data_processor.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    Attributes:
+        _data_processor_type: used to instantiate the correct class object
+        subclasses: dictionnary mapping _data_processor_type to object
+    """
+
+    _data_processor_type: Optional[str] = None
+    subclasses = {}
+
+    def __init_subclass__(cls, *args, **kwargs) -> None:
```
[the remark this answers is not in the document]
Comment (TeddyCr) [reply ingestion/src/metadata/data_insight/processor/data_processor.py:38]: It is just used to instantiate the correct class instance based on the processor type. 🙂

👉 [__init_subclass__](https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__)

[the remark this answers is not in the document]
Comment (pmbrull) [reply ingestion/src/metadata/data_insight/processor/data_processor.py:38]: oooh thanks

**Review by pmbrull — APPROVED** [review]

**Pull request merged**